apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx

```diff
@@ -231,47 +231,57 @@ const createDataObject = (
   };
 };
 
-function calculateDepthCost(
+export function calculateDepthCost(
   pairToken: TokensData,
   poolSide: "in" | "out",
   data: AnalysisData,
   amm: AMM<PoolPairData>,
   poolType: PoolTypeEnum,
 ) {
-  const { analysisToken } = usePoolSimulator();
-
+  const analysisToken = data.tokens.find(
+    (token) => token.symbol !== pairToken.symbol,
+  );
+  if (!analysisToken) throw new Error("Analysis token not found");
```

Comment:
If you did this to not have to use the context, I think that would be best to require the analysis token as an input of the function.

Reply:
you say this in case of more than 2 tokens right?